web/src/settings_notifications.ts

```diff
@@ -171,6 +182,72 @@ function stream_notification_setting_changed(target: HTMLInputElement): void {
     );
 }
 
+function change_state_of_customize_stream_notifications_widget(
+    event: JQuery.ClickEvent,
+    dropdown: tippy.Instance,
+): void {
+    dropdown.hide();
+    event.preventDefault();
+    event.stopPropagation();
+    assert(customize_stream_notifications_widget !== undefined);
+    customize_stream_notifications_widget.render();
+    const stream_id = customize_stream_notifications_widget.value();
+    assert(typeof stream_id === "number");
+
+    const $customizable_stream_notifications_table = $("#customizable_stream_notifications_table");
+
+    $customizable_stream_notifications_table.find("input[type='checkbox']").prop("disabled", false);
+
+    if (!realm.realm_push_notifications_enabled) {
+        $customizable_stream_notifications_table
+            .find("input.push_notifications")
+            .prop("disabled", true);
+    }
+
+    for (const notification_setting of settings_config.stream_specific_notification_settings) {
+        const checked_state = stream_data.receives_notifications(stream_id, notification_setting);
+        $customizable_stream_notifications_table
+            .find(`.${CSS.escape(notification_setting)}`)
+            .prop("checked", checked_state);
+    }
+}
+
+function get_streams_to_customize_notifications(): dropdown_widget.Option[] {
+    const user_id = people.my_current_user_id();
+
+    return stream_data
+        .get_streams_for_user(user_id)
```

Comment:
This includes archived channels as well. I don't think we want to show them here.